### PR TITLE
Refactor Lvis

### PR DESCRIPTION
**Pull request by jedmccaleb**
Now works with our infra

### CODE REVIEW

File: models/mechs/lvis/images_env.go [hunk 1]
```diff
@@ -113,122 +103,37 @@ func (ev *ImagesEnv) Defaults() {
 
 // ImageList returns the list of images -- train or test
 func (ev *ImagesEnv) ImageList() []string {
-	if ev.Test {
```
Comment (etuleu) [inline models/mechs/lvis/images_env.go:116]: Do we not need the test check anymore?

File: models/mechs/lvis/sim.go [hunk 2]
```diff
@@ -0,0 +1,1392 @@
+package main
```
Comment (etuleu) [inline models/mechs/lvis/sim.go:1]: This file moved or something ?

**Review by etuleu — APPROVED** [review]
I looked at the infra stuff and briefly at the lvis code but randy will probably have some opinions on that part specifically

**Comment by rcoreilly** [comment]
Random comments:

* why does config.go use non-go-like ALL_UPPER_CASE ?  the `gConfig` identifier is presumably sufficient to identify the nature of the parameter?

* I'm pretty sure your train / test split is not comparable. the original holds out 2 unique *items* -- the filenames have 2 numbers, separated by the _
```
airplane_001_00001.png
airplane_001_00002.png
...
```

The first number is the "item" number, which is the identity of the 3D model used to render the images, while the last number indexes the particular render of that item.  When we split train / test we do so excluding exactly 2 of each item, all renders thereof.  There are different numbers of items per category, so one fixed percent: .1 is not going to work.  given that there are roughly 10 items per category, holding out 2 / 10 = roughly 20%, but again this is not going to be a constant for each category..

I added this info to slack for future reference.

* also: `GetTest/TrainImages` always uses the last 10% of renders for testing -- this is actually not very "random" because the 3D objects were curated in a popularity-first order, so later objects are likely to be less "good" models by some unknown metric.  This is also why the "plus" models are worse overall -- I had an RA go back and try to find more items from each category, and these tended to be worse exemplars.

File: models/mechs/lvis/images_env.go (hunk 1 above)
Comment (jedmccaleb) [reply models/mechs/lvis/images_env.go:116]: Nope there is a different ImagesEnv for test or train

**Comment by jedmccaleb** [comment]
Upper case is more clear that it is a config and not a normal variable

for the test/train split, does it really matter? Seems like we just want a rough indication. Sort of arbitrary what we use right?

> also: GetTest/TrainImages always uses the last 10% of renders for testing 
Ah right, I meant to shuffle before doing this

**Comment by rcoreilly** [comment]
I think it is a big difference conceptually if you've never seen that particular 3D shape before, vs. you've seen different renders of that shape but never a particular render.  It matters for interpreting the results.  Nevertheless, it is likely that in most cases performance on one measure will be correlated with the other, so at a practical level it probably doesn't matter that much.  Depends on if you actually care about what the model is doing vs. just using it to index performance.  I happen to care about both :)

Also, with the random shuffle, how are you guaranteeing that the same train / test split is used across all MPI nodes?  Just assuming they have the same random seed at that point?  If you don't have the same split, then it is likely that many more  images are seen during training across all the nodes, invalidating the testing.

And depending on the order of processing since initializing the random seed, you may end up with different random splits across different runs, which is why I had saved those .json files, so that it was guaranteed to end up with fully replicable runs for cases where you change something that *shouldn't matter* (to make sure it didn't) and also to eliminate noise in this variable in interpreting parameter / algorithm changes across runs.

Anyway, it is likely that this all works out fine in practice, but it is just more potentially error-prone compared to having a saved split...  For my $, it is worth eliminating any uncertainty in the above issues, but you can live dangerously if you want.

**Pull request merged**